### ISSUE TITLE
Request record from collection to be available offline

### ISSUE THREAD

**Issue by tagrudev** [GroundMeteor/db#70]
Hello using your project in my current one (thanks for the good job btw) I have a question though - Is it possible to request a record from the client side to be available offline. 

ex. I have a collection of projects, that the user subscribe to but I want the user to be able to decide which of them he/she can keep in the browser memory ?

**Comment by raix** [comment]
Hi @tagrudev - so the stuff I'm working on at the moment is conflict resolution and publish/subscriptions.
What you are suggesting is a kind of filter for what data to persist and what to load when online. (like the pattern for app.cache)

The current idea I'm working on is a bit different - but it might solve the problem if done correctly. Its based on the actual data usage using data weights to help ground db keep inside the storage quotas.
If on cordova, the quota problems is not as big as when using local storage, but its still nice to be able to set some quotas/limits.

The app will pr. default try to keep the data most important to the user - I guess the weighting implementation allows the developer to setup custom weight algoritmes eg. you could add a weight param adding weight if the data id was in the project subscription - allowing the user to say "I really want this offline" - but still the data could be discarded if the quota limit is hit.

Makes sense, or am I just too deep in code atm? :)

In short: ground db will try to keep the important data offline pr. default - but you can help it by adding custom weight params.

**Comment by raix** [comment]
btw. for now you could spin up a client only collection and copy the project docs in there

**Comment by tagrudev** [comment]
Hi @raix thanks for the fast response, well I would build the conflict solving myself when there is a change I should provide the user to decide which to keep and which to overwrite but I am not there yet I want to provide the option to decide which one to keep from the subscription :+1: 

So I should clone the whole subscription on a client only collection >? Or am I understanding you wrong ?

**Comment by raix** [comment]
Yes, clone the projects the user wants offline into a separate collection - it takes more in memory though.

(the default conflict handler in new version is "let the newest data win" - current Meteor solution is "the data last on server wins" - its possible to involve the user in case one wants the user to decide etc. and one can of cause build own conflict handlers to plugin)*
